lib/Controller/MessagesController.php

```diff
@@ -307,6 +306,10 @@ public function getThread(int $id): JSONResponse {
 			return new JSONResponse([], Http::STATUS_FORBIDDEN);
 		}
 
+		if (empty($message->getThreadRootId())) {
```

Comment:
in what scenario do we allow the thread root ID to be null? shouldn't it always have a value, worst case it's own message ID, no?

Reply:
Is it?  didn't know. Seems like the issue is cropping up somewhere else then since the related ticket is having issues with exactly that.

Reply:
yeah. because if it's NULL, what thread does it belong to?

ref https://github.com/nextcloud/mail/issues/5747 for one of the more recent incidents where the ID was erroneously null.

Reply:
Right, programatically, we're currently supporting null values for the threadRootId: https://github.com/nextcloud/mail/blob/8aabd4dede1ff2f7021b018e72a9aee35595e268/lib/Db/Message.php#L175

We could change that line to use the `messageId` in case the `threadRootId` is empty? Would prevent future issues like this.

And as for fixing the old ones, a repair step that encapsulates https://github.com/nextcloud/mail/issues/5747#issuecomment-979483114 ?

Reply:
Let's do that!